Zend/zend_ast.h

```diff
@@ -213,6 +213,37 @@ typedef struct _zend_ast_decl {
 	zend_ast *child[5];
 } zend_ast_decl;
 
+typedef union _znode_op {
+	uint32_t      constant;
+	uint32_t      var;
+	uint32_t      num;
+	uint32_t      opline_num; /*  Needs to be signed */
+#if ZEND_USE_ABS_JMP_ADDR
+	zend_op       *jmp_addr;
+#else
+	uint32_t      jmp_offset;
+#endif
+#if ZEND_USE_ABS_CONST_ADDR
+	zval          *zv;
+#endif
+} znode_op;
+
+typedef struct _znode { /* used only during compilation */
+	uint8_t op_type;
+	uint8_t flag;
+	union {
+		znode_op op;
+		zval constant; /* replaced by literal/zv */
+	} u;
+} znode;
```

Comment:
The above are really much more related to compilation than to ast.

Reply:
I moved it here because of the ``/* Temporarily defined here, to avoid header ordering issues */`` comment above ``zend_ast_znode``, and as this depends on ``znode`` I moved everything here. But I can move that back to zend_compile, I don't really care.

Reply:
IMO znode should stay in zend_compile.h.